multimap/src/main/java/org/infinispan/multimap/impl/ListBucket.java

```diff
@@ -96,6 +96,46 @@ public ListBucket<V> offer(V value, boolean first) {
       return new ListBucket<>(newBucket);
    }
 
+   public Collection<V> sublist(long from, long to) {
+      List<V> result = new ArrayList<>();
+      // index request
+      if (from == to) {
+         V element = index(from);
+         if (element != null) {
+            result.add(element);
+         }
+         return result;
+      }
+
+      // from and to are + but from is bigger
+      // example: from 2 > to 1 -> empty result
+      // from and to are - and to is smaller
+      // example: from -1 > to -2 -> empty result
+      if ((from > 0 && to > 0 && from > to) || (from < 0 && to < 0 && from > to)) {
+         return result;
+      }
```

Comment:
Maybe it's worth moving this to be the first check, so we can return Collections.emptyList() and don't allocate the list.